src/AudioFileSourceID3.cpp

```diff
@@ -234,6 +234,7 @@ uint32_t AudioFileSourceID3::read(void *data, uint32_t len)
       }
     }
   } while (!id3.eof());
+  src->setDataPos(src->getPos());
```

Comment:
Could you also add this to the WAVFile class?  That has a header as well.

Reply:
Will do. I figured there were others with headers to do this with, I just hadn't tested any other file types.